src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -767,177 +633,105 @@ namespace winrt::Microsoft::Terminal::Control::implementation
+                                                               panelScaleX,
+                                                               panelScaleY);
+        if (!coreInitialized)
+        {
+            return false;
+        }
+        _interactivity->Initialize();
 
-            auto bottom = _terminal->GetViewport().BottomExclusive();
-            auto bufferHeight = bottom;
+        _AttachDxgiSwapChainToXaml(_core->GetSwapChain());
 
-            ScrollBar().Maximum(bufferHeight - bufferHeight);
-            ScrollBar().Minimum(0);
-            ScrollBar().Value(0);
-            ScrollBar().ViewportSize(bufferHeight);
-            ScrollBar().LargeChange(std::max<SHORT>(bufferHeight - 1, 0)); // scroll one "screenful" at a time when the scroll bar is clicked
+        // Tell the DX Engine to notify us when the swap chain changes. We do
+        // this after we initially set the swapchain so as to avoid unnecessary
+        // callbacks (and locking problems)
+        _core->SwapChainChanged({ get_weak(), &TermControl::RenderEngineSwapChainChanged });
 
-            localPointerToThread->EnablePainting();
+        auto bufferHeight = _core->BufferHeight();
 
-            // Set up blinking cursor
-            int blinkTime = GetCaretBlinkTime();
-            if (blinkTime != INFINITE)
-            {
-                // Create a timer
-                DispatcherTimer cursorTimer;
-                cursorTimer.Interval(std::chrono::milliseconds(blinkTime));
-                cursorTimer.Tick({ get_weak(), &TermControl::_CursorTimerTick });
-                cursorTimer.Start();
-                _cursorTimer.emplace(std::move(cursorTimer));
-            }
-            else
-            {
-                // The user has disabled cursor blinking
-                _cursorTimer = std::nullopt;
-            }
+        ScrollBar().Maximum(bufferHeight - bufferHeight);
+        ScrollBar().Minimum(0);
+        ScrollBar().Value(0);
+        ScrollBar().ViewportSize(bufferHeight);
+        ScrollBar().LargeChange(std::max(bufferHeight - 1, 0)); // scroll one "screenful" at a time when the scroll bar is clicked
 
-            // Set up blinking attributes
-            BOOL animationsEnabled = TRUE;
-            SystemParametersInfoW(SPI_GETCLIENTAREAANIMATION, 0, &animationsEnabled, 0);
-            if (animationsEnabled && blinkTime != INFINITE)
-            {
-                // Create a timer
-                DispatcherTimer blinkTimer;
-                blinkTimer.Interval(std::chrono::milliseconds(blinkTime));
-                blinkTimer.Tick({ get_weak(), &TermControl::_BlinkTimerTick });
-                blinkTimer.Start();
-                _blinkTimer.emplace(std::move(blinkTimer));
-            }
-            else
-            {
-                // The user has disabled blinking
-                _blinkTimer = std::nullopt;
-            }
+        // Mild worry that doing EnablePainting in Core::InitializeTerminal,

```

Comment:
is this mild worry something that should be promoted to a "This is why we don't do this" comment?